dataframe-jdbc/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/h2/h2Test.kt

```diff
@@ -579,6 +660,51 @@ class JdbcTest {
         dataSchema1.columns["name"]!!.type shouldBe typeOf<String?>()
     }
 
+    @Test
+    fun `read from sql query with extension functions`() {
+        @Language("SQL")
+        val sqlQuery = """
+            SELECT c.name as customerName, SUM(s.amount) as totalSalesAmount
+            FROM Sale s
+            INNER JOIN Customer c ON s.customerId = c.id
+            WHERE c.age > 35
+            GROUP BY s.customerId, c.name
+        """.trimIndent()
+
+        val df = connection.toDF(sqlQuery).cast<CustomerSales>()
```

Comment:
Looks similar to File.readDataFrame, what do you think about this name?
https://github.com/Kotlin/dataframe/blob/f631f68cd62ee563356e23943fe4ac883aea959d/core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/guess.kt#L291